test/utils/mockutils/mockserver/reactive.go

```diff
@@ -0,0 +1,61 @@
+package mockserver
+
+import (
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/amp-labs/connectors/test/utils/mockutils"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockcond"
+)
+
+// Reactive is a server recipe that describes how mock server should react when conditions are met.
+type Reactive struct {
```

Comment:
Well, every server is reactive, so can we call this `ConditionalServer` to be more explicit that it handles if/else blocks?